### PR TITLE
Publish apm-agent-python docker images in dockerhub

### PR DESCRIPTION
**Pull request by reakaleek**
## What does this PR do?

Publishes apm-agent-python docker images for testing in dockerhub so it can be accessed publicly in GH Actions workflows.

## Why is it important?

Optimizes the test time

## Related Issues
- depends on https://github.com/elastic/apm-agent-python/pull/1794

### CODE REVIEW

**Review by v1v — COMMENTED** [review]
As far as I see, those python versions are explicitly set in this file, if for instance, python `3.6` reaches EOL or it's removed from  [.matrix_python_full.yml](https://github.com/elastic/apm-agent-python/blob/main/.ci/.matrix_python_full.yml) or if `3.12` is added to that file, then it's required to change this file.
Do you think we can use something more dynamic?

**Comment by reakaleek** [comment]
> As far as I see, those python versions are explicitly set in this file, if for instance, python `3.6` reaches EOL or it's removed from [.matrix_python_full.yml](https://github.com/elastic/apm-agent-python/blob/main/.ci/.matrix_python_full.yml) or if `3.12` is added to that file, then it's required to change this file. Do you think we can use something more dynamic?

Use let me draft something

File: .ci/.docker-images.yml [hunk 1]
```diff
@@ -16,6 +16,8 @@ templates:
     name: "apm-agent-python"
     repository: "elastic/apm-agent-python"
     working_directory: "tests"
+    build_script: docker build --build-arg PYTHON_IMAGE=${TAG/-/:} -t elasticobservability/apm-agent-python:${TAG} .
```
Comment (beniwohli) [inline .ci/.docker-images.yml:19]: Just to make sure nobody mistakes these images for actual release artifacts of the agent itself, we could rename them to something like `apm-agent-python-testing`

**Review by beniwohli — APPROVED** [review]
Awesome!

**Review by v1v — APPROVED** [review]

**Pull request merged**